api/v1alpha1/nnfsystemstorage_types.go

```diff
@@ -0,0 +1,136 @@
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	dwsv1alpha2 "github.com/DataWorkflowServices/dws/api/v1alpha2"
+	"github.com/DataWorkflowServices/dws/utils/updater"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NnfSystemStorageComputesTarget string
+
+const (
+	ComputesTargetAll     NnfSystemStorageComputesTarget = "all"
+	ComputesTargetEven    NnfSystemStorageComputesTarget = "even"
+	ComputesTargetOdd     NnfSystemStorageComputesTarget = "odd"
+	ComputesTargetPattern NnfSystemStorageComputesTarget = "pattern"
+)
+
+// NnfSystemStorageSpec defines the desired state of NnfSystemStorage
+type NnfSystemStorageSpec struct {
+	// SystemConfiguration is an object reference to the SystemConfiguration resource to use. If this
+	// field is empty, name: default namespace: default is used.
+	SystemConfiguration corev1.ObjectReference `json:"systemConfiguration,omitempty"`
+
+	// ExludeRabbits is a list of Rabbits to exclude from the Rabbits in the SystemConfiguration
+	ExcludeRabbits []string `json:"excludeRabbits,omitempty"`
+
+	// IncludeRabbits is a list of Rabbits to use rather than getting the list of Rabbits from the
+	// SystemConfiguration
+	IncludeRabbits []string `json:"includeRabbits,omitempty"`
+
+	// ExcludeComputes is a list of compute nodes to exclude from the the compute nodes listed in the
+	// SystemConfiguration
+	ExcludeComputes []string `json:"excludeComputes,omitempty"`
+
+	// IncludeComputes is a list of computes nodes to use rather than getting the list of compute nodes
+	// from the SystemConfiguration
+	IncludeComputes []string `json:"includeComputes,omitempty"`
+
+	// ComputesTarget specifies which computes to make the storage accessible to
+	// +kubebuilder:validation:Enum=all;even;odd;pattern
+	// +kubebuilder:default:=all
+	ComputesTarget NnfSystemStorageComputesTarget `json:"computesTarget"`
+
+	// ComputesPattern is a list of compute node indexes (0-15) to make the storage accessible to. This
+	// is only used if ComputesTarget is "pattern"
+	// +kubebuilder:validation:MaxItems=16
+	// +kubebuilder:validation:items:Maximum=15
+	// +kubebuilder:validation:items:Minimum=0
+	ComputesPattern []int `json:"computesPattern,omitempty"`
+
+	// Capacity is the allocation size on each Rabbit
+	// +kubebuilder:default:=1073741824
+	Capacity int64 `json:"capacity"`
+
+	// Type is the file system type to use for the storage allocation
+	// +kubebuilder:validation:Enum=raw
```

Comment:
Let's add XFS right away. Then you don't have to fuss with it down the road when you would have to make a conversion routine for it.

Reply:
Okay. I'll do gfs2 as well then.